text/0000-source-date-epoch.md

```diff
@@ -0,0 +1,51 @@
+# Meta
+[meta]: #meta
+- Name: Source Date Epoch
+- Start Date: 2020-01-28
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Allow platforms to define an build epoch. This would improve the UX of images builds at the expense of reproducibility.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the lifecycle sets all build dates to the unix epoch for reproducibility. This can be confusing for newcomers because it looks like their builds are "40 years old". This should at the least be documented. Going one step further, the lifecycle could respect the value of `$SOURCE_DATE_EPOCH` if set by platforms, which would allow an optional override.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Lifecycle respects `$SOURCE_DATE_EPOCH` if provided by platform by using that value as the timestamp for everything it creates. This could have a default value of zero for repro. and be set optionally to allow platforms a better UX. eg. `pack build MY_APP --build-epoch 12345` would set `SOURCE_DATE_EPOCH=12345` during build  time and all layers would have that timestamp.
```

Comment:
> all layers would have that timestamp


Does "layers" refer to layer history entries or does it include layer file timestamps. If layer file timestamps are not standardized, lifecycle cannot reuse cached layers anymore. We currently use the layer diffID (uncompressed layer digest), to determine whether a layer has changed